ts/features/pn/screens/LegacyAttachmentPreviewScreen.tsx

```diff
@@ -29,54 +28,54 @@ export const LegacyAttachmentPreviewScreen = ({
   // This ref is needed otherwise the auto back on the useEffect will fire multiple
   // times, since its dependencies change during the back navigation
   const autoBackOnErrorHandled = useRef(false);
-  const maybePnMessageAttachment = useIOSelector(state =>
+  const pnMessageAttachmentOrUndefined = useIOSelector(state =>
     pnMessageAttachmentSelector(state)(messageId)(attachmentId)
   );
 
   useEffect(() => {
     // This condition happens only if this screen is shown without having
     // first retrieved the third party message (so it should never happen)
-    if (!autoBackOnErrorHandled.current && O.isNone(maybePnMessageAttachment)) {
+    if (!autoBackOnErrorHandled.current && !pnMessageAttachmentOrUndefined) {
       // eslint-disable-next-line functional/immutable-data
       autoBackOnErrorHandled.current = true;
       navigation.goBack();
     }
-  }, [maybePnMessageAttachment, navigation]);
+  }, [pnMessageAttachmentOrUndefined, navigation]);
 
-  return O.isSome(maybePnMessageAttachment) ? (
+  return pnMessageAttachmentOrUndefined ? (
```

Comment:
same here

Reply:
The output type is either an object's instance or undefined so there is no strange JS casting